controllers/dspipeline_controller.go

```
@@ -386,6 +400,36 @@ func (r *DSPAReconciler) setStatus(ctx context.Context, resourceName string, con
+	if err != nil {
+		return false
+	}
+
+	availableCond := util.GetDeploymentCondition(deployment.Status, appsv1.DeploymentAvailable)
+	if availableCond != nil && availableCond.Status == corev1.ConditionTrue {
+		return true
+	}
+	return false
+}
+
+func (r *DSPAReconciler) checkAvailableKubernetesDSPAs(ctx context.Context) (bool, error) {
```

golangci-lint (1.64.8)

func `(*DSPAReconciler).checkAvailableKubernetesDSPAs` is unused

(unused)